deeposlandia/paramoptim.py

```diff
@@ -0,0 +1,298 @@
+
+def run_model(train_generator, validation_generator, output_folder,
+              instance_name, image_size, aggregate_value, nb_labels,
+              nb_training_image, nb_validation_image,
+              batch_size, dropout, network, learning_rate, learning_rate_decay):
+    """
+    """
+    if args.model == "feature_detection":
+        net = FeatureDetectionNetwork(network_name=instance_name,
+                                      image_size=image_size,
+                                      nb_channels=3,
+                                      nb_labels=nb_labels,
+                                      architecture=network)
+        loss_function = "binary_crossentropy"
+    elif args.model == "semantic_segmentation":
+        net = SemanticSegmentationNetwork(network_name=instance_name,
+                                          image_size=image_size,
+                                          nb_channels=3,
+                                          nb_labels=nb_labels,
+                                          architecture=network)
+        loss_function = "categorical_crossentropy"
+    else:
+        utils.logger.error(("Unrecognized model. Please enter 'feature_detection' "
+                            "or 'semantic_segmentation'."))
+        sys.exit(1)
+    model = Model(net.X, net.Y)
+    opt = Adam(lr=learning_rate, decay=learning_rate_decay)
+    model.compile(loss=loss_function, optimizer=opt, metrics=['acc'])
+
+    # Model training
+    STEPS = nb_training_image // batch_size
+    VAL_STEPS = nb_validation_image // batch_size
+    checkpoint_filename = os.path.join(output_folder,
+                                       "checkpoint-epoch-{epoch:03d}.h5")
+    checkpoints = callbacks.ModelCheckpoint(
+        checkpoint_filename,
+        monitor='val_acc',
+        verbose=0,
+        save_best_only=True,
+        save_weights_only=False,
+        mode='auto', period=1)
+    terminate_on_nan = callbacks.TerminateOnNaN()
+    earlystop = callbacks.EarlyStopping(monitor='val_acc',
+                                        min_delta=0.005,
+                                        patience=5,
+                                        verbose=0,
+                                        mode='auto')
+    hist = model.fit_generator(train_generator,
+                               epochs=args.nb_epochs,
+                               steps_per_epoch=STEPS,
+                               validation_data=validation_generator,
+                               validation_steps=VAL_STEPS,
+                               callbacks=[checkpoints, earlystop, terminate_on_nan])
+    ref_metric = max(hist.history['val_acc'])
+    return {'model': model, 'val_acc': ref_metric,
+            'batch_size': batch_size, 'network': network, 'dropout': dropout,
+            'learning_rate': learning_rate, 'learning_rate_decay': learning_rate_decay}
+
+if __name__=='__main__':
+    """Main method: run a convolutional neural network using Keras API
```

Comment:
Maybe turn the string into a comment. This is not a function docstring.

Reply:
Or you can move this content into a really `main` function.

Reply:
I've finally dropped it: the information already appears in the module docstring.